File: sdkwork-app-sdk-typescript/src/api/vip.ts
import { appApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { PlusApiResultBoolean, PlusApiResultInviteRulesVO, PlusApiResultListVipBenefitVO, PlusApiResultListVipCouponVO, PlusApiResultListVipInviteRecordVO, PlusApiResultListVipLevelVO, PlusApiResultListVipPackGroupVO, PlusApiResultListVipPackVO, PlusApiResultListVipPointsHistoryVO, PlusApiResultLong, PlusApiResultVipDailyRewardStatusVO, PlusApiResultVipDailyRewardVO, PlusApiResultVipInfoVO, PlusApiResultVipInviteInfoVO, PlusApiResultVipInviteVO, PlusApiResultVipPackDetailVO, PlusApiResultVipPackGroupDetailVO, PlusApiResultVipPrivilegeUsageVO, PlusApiResultVipPurchaseVO, PlusApiResultVipStatusVO, PlusApiResultVoid, SpeedUpForm, VipInviteForm, VipPurchaseForm, VipRenewForm, VipUpgradeForm } from '../types';


export class VipApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** 购买VIP */
  async purchase(body: VipPurchaseForm): Promise<PlusApiResultVipPurchaseVO> {
    return this.client.post<PlusApiResultVipPurchaseVO>(appApiPath(`/vip/purchase`), body);
  }

/** 升级VIP */
  async upgrade(body: VipUpgradeForm): Promise<PlusApiResultVipPurchaseVO> {
    return this.client.post<PlusApiResultVipPurchaseVO>(appApiPath(`/vip/purchase/upgrade`), body);
  }

/** 续费VIP */
  async renew(body: VipRenewForm): Promise<PlusApiResultVipPurchaseVO> {
    return this.client.post<PlusApiResultVipPurchaseVO>(appApiPath(`/vip/purchase/renew`), body);
  }

/** 使用加速特权 */
  async useSpeedUpPrivilege(body: SpeedUpForm): Promise<PlusApiResultVoid> {
    return this.client.post<PlusApiResultVoid>(appApiPath(`/vip/privilege/speed-up`), body);
  }

/** 领取每日奖励 */
  async claimDailyReward(): Promise<PlusApiResultVipDailyRewardVO> {
    return this.client.post<PlusApiResultVipDailyRewardVO>(appApiPath(`/vip/points/daily-reward`));
  }

/** 邀请好友 */
  async inviteFriend(body: VipInviteForm): Promise<PlusApiResultVipInviteVO> {
    return this.client.post<PlusApiResultVipInviteVO>(appApiPath(`/vip/invite`), body);
  }

/** 领取优惠券 */
  async claimCoupon(couponId: string | number): Promise<PlusApiResultVoid> {
    return this.client.post<PlusApiResultVoid>(appApiPath(`/vip/coupons/${couponId}/claim`));
  }

/** 获取VIP状态 */
  async getVipStatus(): Promise<PlusApiResultVipStatusVO> {
    return this.client.get<PlusApiResultVipStatusVO>(appApiPath(`/vip/status`));
  }

/** 获取特权使用情况 */
  async getPrivilegeUsage(): Promise<PlusApiResultVipPrivilegeUsageVO> {
    return this.client.get<PlusApiResultVipPrivilegeUsageVO>(appApiPath(`/vip/privilege/usage`));
  }

/** 获取积分明细 */
  async getPointsHistory(): Promise<PlusApiResultListVipPointsHistoryVO> {
    return this.client.get<PlusApiResultListVipPointsHistoryVO>(appApiPath(`/vip/points/history`));
  }

/** 获取每日奖励状态 */
  async getDailyRewardStatus(): Promise<PlusApiResultVipDailyRewardStatusVO> {
    return this.client.get<PlusApiResultVipDailyRewardStatusVO>(appApiPath(`/vip/points/daily-reward/status`));
  }

/** 获取积分余额 */
  async getPointsBalance(): Promise<PlusApiResultLong> {
    return this.client.get<PlusApiResultLong>(appApiPath(`/vip/points/balance`));
  }

/** 获取套餐分组列表 */
  async listPackGroups(params?: QueryParams): Promise<PlusApiResultListVipPackGroupVO> {
    return this.client.get<PlusApiResultListVipPackGroupVO>(appApiPath(`/vip/pack-groups`), params);
  }

/** 获取分组详情 */
  async getPackGroupDetail(groupId: string | number): Promise<PlusApiResultVipPackGroupDetailVO> {
    return this.client.get<PlusApiResultVipPackGroupDetailVO>(appApiPath(`/vip/pack-groups/${groupId}`));
  }

/** 获取分组套餐 */
  async listPacksByGroup(groupId: string | number): Promise<PlusApiResultListVipPackVO> {
    return this.client.get<PlusApiResultListVipPackVO>(appApiPath(`/vip/pack-groups/${groupId}/packs`));
  }

/** 获取所有套餐 */
  async listAllPacks(params?: QueryParams): Promise<PlusApiResultListVipPackVO> {
    return this.client.get<PlusApiResultListVipPackVO>(appApiPath(`/vip/pack-groups/packs`), params);
  }

/** 获取套餐详情 */
  async getPackDetail(packId: string | number): Promise<PlusApiResultVipPackDetailVO> {
    return this.client.get<PlusApiResultVipPackDetailVO>(appApiPath(`/vip/pack-groups/packs/${packId}`));
  }

/** 对比套餐 */
  async comparePacks(params?: QueryParams): Promise<PlusApiResultListVipPackGroupVO> {
    return this.client.get<PlusApiResultListVipPackGroupVO>(appApiPath(`/vip/pack-groups/compare`), params);
  }

/** 获取VIP等级列表 */
  async listVipLevels(): Promise<PlusApiResultListVipLevelVO> {
    return this.client.get<PlusApiResultListVipLevelVO>(appApiPath(`/vip/levels`));
  }

/** 获取邀请规则 */
  async getInviteRules(): Promise<PlusApiResultInviteRulesVO> {
    return this.client.get<PlusApiResultInviteRulesVO>(appApiPath(`/vip/invite/rules`));
  }

/** 获取邀请记录 */
  async getInviteRecords(params?: QueryParams): Promise<PlusApiResultListVipInviteRecordVO> {
    return this.client.get<PlusApiResultListVipInviteRecordVO>(appApiPath(`/vip/invite/records`), params);
  }

/** 获取邀请信息 */
  async getInviteInfo(): Promise<PlusApiResultVipInviteInfoVO> {
    return this.client.get<PlusApiResultVipInviteInfoVO>(appApiPath(`/vip/invite/info`));
  }

/** 获取VIP信息 */
  async getVipInfo(): Promise<PlusApiResultVipInfoVO> {
    return this.client.get<PlusApiResultVipInfoVO>(appApiPath(`/vip/info`));
  }

/** 获取VIP优惠券 */
  async listVipCoupons(): Promise<PlusApiResultListVipCouponVO> {
    return this.client.get<PlusApiResultListVipCouponVO>(appApiPath(`/vip/coupons`));
  }

/** 获取我的优惠券 */
  async listMyCoupons(params?: QueryParams): Promise<PlusApiResultListVipCouponVO> {
    return this.client.get<PlusApiResultListVipCouponVO>(appApiPath(`/vip/coupons/my`), params);
  }

/** 检查VIP状态 */
  async checkVipStatus(): Promise<PlusApiResultBoolean> {
    return this.client.get<PlusApiResultBoolean>(appApiPath(`/vip/check`));
  }

/** 获取VIP权益 */
  async listVipBenefits(params?: QueryParams): Promise<PlusApiResultListVipBenefitVO> {
    return this.client.get<PlusApiResultListVipBenefitVO>(appApiPath(`/vip/benefits`), params);
  }
}

export function createVipApi(client: HttpClient): VipApi {
  return new VipApi(client);
}

File: sdkwork-app-sdk-go/api/vip.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/app-sdk/types"
    sdkhttp "github.com/sdkwork/app-sdk/http"
)

type VipApi struct {
    client *sdkhttp.Client
}

func NewVipApi(client *sdkhttp.Client) *VipApi {
    return &VipApi{client: client}
}

// 购买VIP
func (a *VipApi) Purchase(body sdktypes.VipPurchaseForm) (sdktypes.PlusApiResultVipPurchaseVO, error) {
    raw, err := a.client.Post(AppApiPath("/vip/purchase"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultVipPurchaseVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultVipPurchaseVO](raw)
}

// 升级VIP
func (a *VipApi) Upgrade(body sdktypes.VipUpgradeForm) (sdktypes.PlusApiResultVipPurchaseVO, error) {
    raw, err := a.client.Post(AppApiPath("/vip/purchase/upgrade"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultVipPurchaseVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultVipPurchaseVO](raw)
}

// 续费VIP
func (a *VipApi) Renew(body sdktypes.VipRenewForm) (sdktypes.PlusApiResultVipPurchaseVO, error) {
    raw, err := a.client.Post(AppApiPath("/vip/purchase/renew"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultVipPurchaseVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultVipPurchaseVO](raw)
}

// 使用加速特权
func (a *VipApi) UseSpeedUpPrivilege(body sdktypes.SpeedUpForm) (sdktypes.PlusApiResultVoid, error) {
    raw, err := a.client.Post(AppApiPath("/vip/privilege/speed-up"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultVoid
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultVoid](raw)
}

// 领取每日奖励
func (a *VipApi) ClaimDailyReward() (sdktypes.PlusApiResultVipDailyRewardVO, error) {
    raw, err := a.client.Post(AppApiPath("/vip/points/daily-reward"), nil, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultVipDailyRewardVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultVipDailyRewardVO](raw)
}

// 邀请好友
func (a *VipApi) InviteFriend(body sdktypes.VipInviteForm) (sdktypes.PlusApiResultVipInviteVO, error) {
    raw, err := a.client.Post(AppApiPath("/vip/invite"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultVipInviteVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultVipInviteVO](raw)
}

// 领取优惠券
func (a *VipApi) ClaimCoupon(couponId string) (sdktypes.PlusApiResultVoid, error) {
    raw, err := a.client.Post(AppApiPath(fmt.Sprintf("/vip/coupons/%s/claim", couponId)), nil, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultVoid
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultVoid](raw)
}

// 获取VIP状态
func (a *VipApi) GetVipStatus() (sdktypes.PlusApiResultVipStatusVO, error) {
    raw, err := a.client.Get(AppApiPath("/vip/status"), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultVipStatusVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultVipStatusVO](raw)
}

// 获取特权使用情况
func (a *VipApi) GetPrivilegeUsage() (sdktypes.PlusApiResultVipPrivilegeUsageVO, error) {
    raw, err := a.client.Get(AppApiPath("/vip/privilege/usage"), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultVipPrivilegeUsageVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultVipPrivilegeUsageVO](raw)
}

// 获取积分明细
func (a *VipApi) GetPointsHistory() (sdktypes.PlusApiResultListVipPointsHistoryVO, error) {
    raw, err := a.client.Get(AppApiPath("/vip/points/history"), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultListVipPointsHistoryVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListVipPointsHistoryVO](raw)
}

// 获取每日奖励状态
func (a *VipApi) GetDailyRewardStatus() (sdktypes.PlusApiResultVipDailyRewardStatusVO, error) {
    raw, err := a.client.Get(AppApiPath("/vip/points/daily-reward/status"), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultVipDailyRewardStatusVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultVipDailyRewardStatusVO](raw)
}

// 获取积分余额
func (a *VipApi) GetPointsBalance() (sdktypes.PlusApiResultLong, error) {
    raw, err := a.client.Get(AppApiPath("/vip/points/balance"), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultLong
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultLong](raw)
}

// 获取套餐分组列表
func (a *VipApi) ListPackGroups(query map[string]interface{}) (sdktypes.PlusApiResultListVipPackGroupVO, error) {
    raw, err := a.client.Get(AppApiPath("/vip/pack-groups"), query, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultListVipPackGroupVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListVipPackGroupVO](raw)
}

// 获取分组详情
func (a *VipApi) GetPackGroupDetail(groupId string) (sdktypes.PlusApiResultVipPackGroupDetailVO, error) {
    raw, err := a.client.Get(AppApiPath(fmt.Sprintf("/vip/pack-groups/%s", groupId)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultVipPackGroupDetailVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultVipPackGroupDetailVO](raw)
}

// 获取分组套餐
func (a *VipApi) ListPacksByGroup(groupId string) (sdktypes.PlusApiResultListVipPackVO, error) {
    raw, err := a.client.Get(AppApiPath(fmt.Sprintf("/vip/pack-groups/%s/packs", groupId)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultListVipPackVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListVipPackVO](raw)
}

// 获取所有套餐
func (a *VipApi) ListAllPacks(query map[string]interface{}) (sdktypes.PlusApiResultListVipPackVO, error) {
    raw, err := a.client.Get(AppApiPath("/vip/pack-groups/packs"), query, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultListVipPackVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListVipPackVO](raw)
}

// 获取套餐详情
func (a *VipApi) GetPackDetail(packId string) (sdktypes.PlusApiResultVipPackDetailVO, error) {
    raw, err := a.client.Get(AppApiPath(fmt.Sprintf("/vip/pack-groups/packs/%s", packId)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultVipPackDetailVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultVipPackDetailVO](raw)
}

// 对比套餐
func (a *VipApi) ComparePacks(query map[string]interface{}) (sdktypes.PlusApiResultListVipPackGroupVO, error) {
    raw, err := a.client.Get(AppApiPath("/vip/pack-groups/compare"), query, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultListVipPackGroupVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListVipPackGroupVO](raw)
}

// 获取VIP等级列表
func (a *VipApi) ListVipLevels() (sdktypes.PlusApiResultListVipLevelVO, error) {
    raw, err := a.client.Get(AppApiPath("/vip/levels"), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultListVipLevelVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListVipLevelVO](raw)
}

// 获取邀请规则
func (a *VipApi) GetInviteRules() (sdktypes.PlusApiResultInviteRulesVO, error) {
    raw, err := a.client.Get(AppApiPath("/vip/invite/rules"), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultInviteRulesVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultInviteRulesVO](raw)
}

// 获取邀请记录
func (a *VipApi) GetInviteRecords(query map[string]interface{}) (sdktypes.PlusApiResultListVipInviteRecordVO, error) {
    raw, err := a.client.Get(AppApiPath("/vip/invite/records"), query, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultListVipInviteRecordVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListVipInviteRecordVO](raw)
}

// 获取邀请信息
func (a *VipApi) GetInviteInfo() (sdktypes.PlusApiResultVipInviteInfoVO, error) {
    raw, err := a.client.Get(AppApiPath("/vip/invite/info"), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultVipInviteInfoVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultVipInviteInfoVO](raw)
}

// 获取VIP信息
func (a *VipApi) GetVipInfo() (sdktypes.PlusApiResultVipInfoVO, error) {
    raw, err := a.client.Get(AppApiPath("/vip/info"), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultVipInfoVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultVipInfoVO](raw)
}

// 获取VIP优惠券
func (a *VipApi) ListVipCoupons() (sdktypes.PlusApiResultListVipCouponVO, error) {
    raw, err := a.client.Get(AppApiPath("/vip/coupons"), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultListVipCouponVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListVipCouponVO](raw)
}

// 获取我的优惠券
func (a *VipApi) ListMyCoupons(query map[string]interface{}) (sdktypes.PlusApiResultListVipCouponVO, error) {
    raw, err := a.client.Get(AppApiPath("/vip/coupons/my"), query, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultListVipCouponVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListVipCouponVO](raw)
}

// 检查VIP状态
func (a *VipApi) CheckVipStatus() (sdktypes.PlusApiResultBoolean, error) {
    raw, err := a.client.Get(AppApiPath("/vip/check"), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultBoolean
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultBoolean](raw)
}

// 获取VIP权益
func (a *VipApi) ListVipBenefits(query map[string]interface{}) (sdktypes.PlusApiResultListVipBenefitVO, error) {
    raw, err := a.client.Get(AppApiPath("/vip/benefits"), query, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultListVipBenefitVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListVipBenefitVO](raw)
}
